exec/cnex/lib/process_win32.c

```diff
@@ -9,6 +9,30 @@
 #include "nstring.h"
 
 
+struct ThreadInfo {
+    HANDLE pHandle;
+    Cell *buffer;
+};
+
+static DWORD WINAPI reader_thread(LPVOID lpParam)
+{
+    struct ThreadInfo *ti = lpParam;
+
+    while (ti->pHandle != INVALID_HANDLE_VALUE) {
+        char buf[1024];
+        DWORD n;
+        if (ti->pHandle != INVALID_HANDLE_VALUE) {
+            if (!ReadFile(ti->pHandle, buf, sizeof(buf), &n, NULL)) {
+                break;
+            }
+            ti->buffer->string = string_appendData(ti->buffer->string, buf, n);
+        }
+    }
+    if (ti->pHandle != INVALID_HANDLE_VALUE) {
+        CloseHandle(ti->pHandle);
```

Comment:
It should probably be the responsibility of the main thread to close the pipe handle, because the main thread is the one that created it.

Reply:
Fixed.